### DISCUSSION TITLE
Is it possible to get user roles from a DM interaction?

### DISCUSSION THREAD

**Discussion by markamoros** [discordjs/discord.js#6612] [Q&A] [answered]
I created a bot that:

1. Listens for DM interactions. 
2. Get the interaction.user.id and fetch the user from the guild with 
`        const user = await server.members.fetch(userId);
`
I am checking the user roles with user.roles.cache but is not getting correctly updated. I get the same roles after updating them from the discord app.

Thank you!

**Comment by SpaceEEC** [comment, marked as answer]
Yes, it is possible:
When fetching the member through [`GuildMemberManager#fetch`](https://discord.js.org/#/docs/main/stable/class/GuildMemberManager?scrollTo=fetch) (as you already do), specify [`FetchMemberOptions`](https://discord.js.org/#/docs/main/stable/typedef/FetchMemberOptions) instead to skip the cache check:
`...fetch({ user: userId, force: true });`
